core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -112,6 +111,16 @@
         Setting.byteSizeSetting("index.translog.flush_threshold_size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    /**
+     * The maximum size of a translog generation. This is independent of the maximum size of
+     * translog operations that have not been flushed.
+     */
+    public static final Setting<ByteSizeValue> INDEX_TRANSLOG_GENERATION_THRESHOLD_SIZE_SETTING =
+            Setting.byteSizeSetting(
+                    "index.translog.generation_threshold_size",
```

Comment:
I pushed dd14ba85cec849219bd8662dadedc166d8d456b6 to move the control of rolling to index shard. Would @nik9000, @bleskes, and @s1monw please take another look?